File: Domain/Entities/Users/User.entity.cs
namespace Domain.Entities.Users;

public record User() : IUser
{
    public required string Id { get; init; }
    public required string Email { get; init; }
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public bool IsEmailConfirmed { get; init; }
}

File: ClientApp/src/app/infrastructure/models/entities/user/user.entity.ts
import { IUser } from '@domain/models/interfaces';
import { Expose } from 'class-transformer';

export class User implements IUser {
  @Expose()
  firstName: string;

  @Expose()
  lastName: string;

  @Expose()
  token: string;

  constructor(firstName: string, lastName: string, token: string) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.token = token;
  }
}
